### DISCUSSION TITLE
A link to a 'basic' page from a 'public' page

### DISCUSSION THREAD

**Discussion by mielune** [Budibase/budibase#8234] [Help] [answered]
Hi budibasers,

I have a usecase that seems to not work using basic features, perhaps you have a hack to do this ;)

- I made a `/welcome` screen, home for `public` role to say 'hello, you are here cool ! click next button to go to the login page'.
- The button action it `Navigate to` to the screen `/main` allowed for `basic` users (and there home screen too).
- Click on the button, the url blink to `#/main` and go back to the public home screen `/welcome`

I would like this public screen to welcome anybody and the rest of the screens protected by the auth process with a button to go to this part of the app from the welcome page.

Have you any way to do this ?

Kind regards,

**Comment by melohagan** [comment]
Hey @mielune 

You can hide links and buttons based on role:
 - https://docs.budibase.com/docs/conditions#showing-and-hiding-a-component-based-on-the-users-role
 - https://docs.budibase.com/docs/navigation#configure-links

**Comment by mielune** [comment, marked as answer]
Hi @melohagan 

Thank you for your answer; but it is not this case.

I don't want to hide the link; My use case is this one :

1. The user connect to a _public screen_ `/welcome` without authentication
2. From this page, the user click a button to "navigate to" `/main`
3. `/main` is for _basic users_ so I would like the user to be redirected to the login flow and then go back to `/main` has a basic role

Currently it don't work like this : when the user (public) click on the button, the application router doesn't find `/main` has a valide route for a public profile and redirect to the public home screen `/welcome`.


Kind regards,

**Reply by melohagan** [reply]
I understand, thanks for the clarification! I did see that come up before: https://github.com/Budibase/budibase/discussions/6796
A subsequent issue was raised here: https://github.com/Budibase/budibase/issues/6798

When I looked into this at the time I was hoping to implement a redirect through a link to the auth page, but this wasn't possible because I think there's more going on with the auth, either cookies or headers - can't remember!
